src/env.h

```diff
@@ -266,6 +266,13 @@ struct node_ares_task {
   RB_ENTRY(node_ares_task) node;
 };
 
+// 4096 bytes on x64 and 4088 on ia32.
+struct node_destroy_ids_list {
+  size_t length;
+  node_destroy_ids_list* prev;
+  int64_t ids[510];
+};
```

Comment:
It's simpler than hand-rolling a vector-like class.  If you're worried about excessive copying, you could simply vec.reserve(512) at startup.